jupyterlab_git/git.py

```diff
@@ -286,6 +287,21 @@ async def clone(self, path, repo_url, auth=None):
             if auth.get("cache_credentials"):
                 await self.ensure_credential_helper(path)
             env["GIT_TERMINAL_PROMPT"] = "1"
+
+            if meta_data_checked:
+                current_content = set(os.listdir(path))
+                code, output, error = await execute(
+                    ["git", "clone", "--depth=1", unquote(repo_url), "-q"],
+                    username=auth["username"],
+                    password=auth["password"],
+                    cwd=path,
+                    env=env,
+                )
+                new_content = set(os.listdir(path))
+                directory = (new_content - current_content).pop()
+                # Check that the directory contains a '.git' folder.
+                shutil.rmtree(f"{directory}/.git")
+
             code, output, error = await execute(
```

Comment:
This needs to be under a `else` condition that will be executed only if `meta_data_checked` is `False`. I guess this is the reason you are still having a .git folder as you clone twice the repo.